src/domain/navigation/index.js

```diff
@@ -132,10 +132,12 @@ export function parseUrlPath(urlPath, currentNavPath, defaultSessionId) {
                 segments.push(roomsSegmentWithRoom(rooms, roomId, currentNavPath));
             }
             segments.push(new Segment("room", roomId));
-            if (currentNavPath.get("details")?.value) {
-                pushRightPanelSegment(segments, "details");
-            } else if (currentNavPath.get("members")?.value) {
-                pushRightPanelSegment(segments, "members");
+            // Add right-panel segments from previous path
+            const previousSegments = currentNavPath.segments;
+            const i = previousSegments.findIndex(s => s.type === "right-panel");
+            if (i !== -1) {
+                segments.push(previousSegments[i]);
```

Comment:
why not just copy over all segments over after right-panel?

Reply:
ah, you wouldn't want to copy the member segment over as we can't assume that the other room will have the same member...

Reply:
will this actually just copy over the members segment but not the member segment?

Reply:
>why not just copy over all segments over after right-panel?

Good idea! We currently only have one segment after the "right-panel" segment but in future we may have more.